src/reconstruction/refinement/updateMetNames.m

```diff
@@ -17,14 +17,22 @@
 % .. Authors:
 %       - written by Diana El Assal 27/06/2017
 
-indMets = findMetIDs(originalModel, modelToUpdate.mets);
+metsReference = strtok(referenceModel.mets, '[');
+metsReference = metsReference(~cellfun('isempty',metsReference));
+metsReference = [metsReference, referenceModel.metNames];
+
+metsUpdate = strtok(modelToUpdate.mets, '[');
+metsUpdate = metsUpdate(~cellfun('isempty',metsUpdate));
+
```

Comment:
As a suggestion:
```
%remove the compartment info
metsReference = regexprep(referenceModel.mets,'\[[^]\]$','');
metsUpdate = regexprep(modelToUpdate.mets,'\[[^]\]$','');
updatedModel = modelToUpdate;
[metsToUpdate,positionsToUpdateFrom] = ismember(metsUpdate,metsReference);

%reset metNames field in modelToUpdate (if requested), or initialize if not existing
if wipeExisting || ~isfield(updatedModel,'metNames')
  updatedModel.metNames = cell(size(updatedModel.mets));
  updatedModel.metNames(~metsToUpdate) = {''}; % init metabolites not present in the second model.
end

updatedModel.metNames(metsToUpdate) = referenceModel.metNames(positionsToUpdateFrom(metsToUpdate));
```
In the current version, if you have any metabolite without a compartment id, the code will fail, as you remove them with the ~cellfun command. The regexp will just remove it.